tests/core/query/test_relational_query.py

```diff
@@ -23,6 +23,110 @@ def test_match_table_to_element(sdata_query_aggregation):
     # TODO: add tests for labels
 
 
+def test_left_inner_right_exclusive_join(sdata_query_aggregation):
+    element_dict, table = join_sdata_spatialelement_table(
+        sdata_query_aggregation, "values_polygons", "table", "right_exclusive"
+    )
+    assert table is None
+    assert all(element_dict[key] is None for key in element_dict)
+
+    sdata_query_aggregation["values_polygons"] = sdata_query_aggregation["values_polygons"].drop([10, 11])
+    with pytest.raises(AssertionError, match="No table with"):
+        join_sdata_spatialelement_table(sdata_query_aggregation, "values_polygons", "not_existing_table", "left")
+
+    # Should we reindex before returning the table?
```

Comment:
Very important point. I think that we should never reindex the geometries (and it doesn't matter if we reindex the table because we never use `table.obs.index` anyway as we use the `region_key` and `instance_key` columns).

Instead, we should definitely match the order of the returned data. I propose an argument: `match_rows_order: bool`. (before I though of making it a string with values `left`, `right`, but thinking about it `bool` is better).

The behavior would be the following: 
- if `False` the order of the returned objects don't change. 
- If `True` and we have a inner join, then a warning is raised that the argument is set to `False`, and the user needs to call, after the inner join, a left or right join with `match_rows_order=True`.
- If `True` and we have a left/right join, then the order is inherited from the left/right.
- If `True` and we have a left/right exclusive join, then we set it to `False` and raise a warning saying that since we have an exclusive join there is no order to match.

Reply:
Actually, not sure if the above is ideal. The most common use case for the user would be a "inner" join, followed by a "right" or "left" join with `match_table_order=True`. So either we allow `match_rows_order` take value `'left'`, `'right'`,`'no'`, either we make the function `match_table_to_element` do exactly this.

Reply:
Iterating more on this. Actually after reviewing the rest of the PR I would actually have `match_rows_order: bool`, with default value `True` and in the case of a inner join return the table sorted to match the elements.

This means:
- with the default value for `match_rows_order`, the exclusive join operations will throw a warning; fine, this will occur rarely because I expect the exclusive joins not to be used often, and the user can always silence it by passing `False`.
- if the user whats to match the order of the shapes to match the table for the inner join, this needs to be done by calling the left join after calling the inner join. A bit tricky, so I would still add teh function `match_table_to_element()`. Note that this operation doesn't make sense when there are multiple elements, so we can restrict `match_table_to_element()` to take a single element instead of a list of them.

Reply:
hmm would we really have to throw a warning in case of an exclusive join for something that is by default defined? We can overwrite in the exclusive join function as standard.

Reply:
Otherwise we can use `match_rows_order: Literal['left', 'right', 'no']` with default value 'no' or 'left'. 

The function `match_element_to_table()` will have the same argument with the same default value.

Reply:
implemented